modules/sc-mesh-secure-deployment/src/2_0/features/cbma/auth/authClient.py

```diff
@@ -0,0 +1,127 @@
+import socket
+import ssl
+import sys
+sys.path.insert(0, '../')
+from tools.verification_tools import *
+from tools.custom_logger import CustomLogger
+from tools.utils import mac_to_ipv6, get_mac_addr
+import glob
+import random
+import time
+
+MAX_RETRIES = 5
+MIN_WAIT_TIME = 1  # seconds
+MAX_WAIT_TIME = 3  # seconds
+
+logger_instance = CustomLogger("authClient")
+
+
+
+class AuthClient:
+    def __init__(self, interface, server_mac, server_port, cert_path, ca_path, mua):
+        self.sslServerIP = mac_to_ipv6(server_mac)
+        self.sslServerPort = server_port
+        self.CERT_PATH = cert_path
+        self.interface = interface
+        self.secure_client_socket = None
+        self.logger = logger_instance.get_logger()
+        self.ca = ca_path
+        self.mymac = get_mac_addr(self.interface)
+        self.server_mac = server_mac
+        self.mua = mua
+
+    def establish_connection(self):
+        # Create an SSL context
+        context = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)
```

Comment:
Since we control both client and server we can use TLS 1.3 which is more secure and even faster 🙂

Python does not seem to make it easy though, as I can't find a ssl.PROTOCOL_TLSv1_3 in https://docs.python.org/3/library/ssl.html and according to only a bit of googling the ssl.PROTOCOL_TLSv* keywords are deprecated.

The way to use TLS1.3+ in Python is apparently:

```
context = ssl.SSLContext(ssl.PROTOCOL_TLS)
context.options |= (ssl.OP_NO_TLSv1 | ssl.OP_NO_TLSv1_1 | ssl.OP_NO_TLSv1_2)
```